src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,86 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int GROW_FACTOR = 2;
+    private Node<K,V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        this.table = new Node[DEFAULT_INITIAL_CAPACITY];;
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= (table.length * DEFAULT_LOAD_FACTOR)) {
+            resize();
+        }
+        int hash = hash(key);
+        Node<K, V> nodeToPut = new Node(key, value);
+        Node<K, V> node = table[hash];
+        if (node == null) {
+            table[hash] = nodeToPut;
+        } else {
+            while (node != null) {
+                if (Objects.equals(key, node.key)) {
+                    node.value = value;
+                    return;
+                }
+                if (node.next == null) {
+                    break;
+                }
```

Comment:
is this 'if' really necessary?

Reply:
I need an exit condition when traversing the bucket